privacy-protections/fingerprinting/helpers/tests.js

```diff
@@ -0,0 +1,1292 @@
+                if (testedProps.includes(propName)) {
+                    return;
+                }
+
+                const propType = typeof screen[propName];
+
+                // ignore props of complex types
+                if (ignoredTypes.includes(propType)) {
+                    return;
+                }
+
+                result[propName] = screen[propName];
+            });
+
+            return result;
+        }
+    },
+
+    // optional - events
+    {
+        id: 'deviceorientation',
+        category: 'events',
+        getValue: () => {
+            let resolve, reject;
+            const promise = new Promise((res, rej) => {resolve = res; reject = rej});
+
+            window.addEventListener('deviceorientation', (event) => {
+                resolve({
+                    alpha: event.alpha,
+                    beta: event.beta,
+                    gamma: event.gamma
+                });
+            });
+
+            return promise;
+        }
+    },
+    {
+        id: 'devicemotion',
+        category: 'events',
+        getValue: () => {
+            let resolve, reject;
+            const promise = new Promise((res, rej) => {resolve = res; reject = rej});
+
+            window.addEventListener('devicemotion', (event) => {
+                resolve({
+                    accelerationX: event.acceleration.x,
+                    accelerationY: event.acceleration.y,
+                    accelerationZ: event.acceleration.z,
+                    rotationRate: event.rotationRate,
+                    interval: event.interval
+                });
+            });
+
+            return promise;
+        }
+    },
+    {
+        id: 'wheel',
+        category: 'events',
+        getValue: () => {
+            let resolve, reject;
+            const promise = new Promise((res, rej) => {resolve = res; reject = rej});
+
+            window.addEventListener('wheel', (event) => {
+                resolve({
+                    deltaX: event.deltaX,
+                    deltaY: event.deltaY,
+                    deltaZ: event.deltaZ,
+                    deltaMode: event.deltaMode
+                });
+            });
+
+            return promise;
+        }
+    },
+    {
+        id: 'touchstart',
+        category: 'events',
+        getValue: () => {
+            let resolve, reject;
+            const promise = new Promise((res, rej) => {resolve = res; reject = rej});
+
+            window.addEventListener('touchstart', (event) => {
+                resolve({
+                    length: event.touches.length,
+                    radiusX: event.touches[0].radiusX,
+                    radiusY: event.touches[0].radiusY,
+                    rotationAngle: event.touches[0].rotationAngle,
+                    force: event.touches[0].force
+                });
+            });
+
+            return promise;
+        }
+    },
+    {
+        id: 'ondevicelight',
+        category: 'events',
```

Comment:
I have not included 'resize' here as the resize event itself doesn't seem to have any additional data (sth like `event.change`). So in order to test resize one can just resize the window and run test again. LMK if I'm missing something here.